trin-core/src/portalnet/overlay.rs

```diff
@@ -69,6 +70,24 @@ impl Default for OverlayConfig {
     }
 }
 
+#[derive(Error, Debug)]
+pub enum SendPingError {
+    #[error("The request timed out")]
+    Timeout,
+
+    #[error("Internal error: ")]
+    Other(discv5::RequestError),
+}
```

Comment:
I'm not sure why we would need a specific error type for a particular request type (e.g. ping). I think we can generalize this error to something like the `discv5::RequestError`. I don't see why we need our own timeout error when the `discv5` enum already defines that variant, and when we do not have any of our own timeout logic. I think it would be better to start with the existing `discv5::RequestError` or define a similar general request error enum that wraps the `discv5::RequestError`. As we encounter overlay-specific error types, we can add them to the enum.

Reply:
I appreciate you taking the time to write out the thorough justification, and I should have taken more time to consider implementation implications. Taking another look at the `discv5::RequestError` enum, I overestimated the number of variants that would be meaningful to consumers of the "send" APIs. And this matching does get quite cumbersome with wrapped errors. While I think there may be other variants that are meaningful to the API consumer (e.g. `InvalidRemoteEnr`), this looks like a good start.

Reply:
No worries :) writing it all out was a good exercise, I probably should have added this rationale to the original PR!